### PR TITLE
Release v0.8.4

### PR DESCRIPTION
**Pull request by bsbodden**
```
fix: repository updateField ignores registered Gson TypeAdapters (resolves gh-284) 
fix: serialize java.time.Instant as milliseconds (resolves gh-275) 
refactor: delombok all non-test sources (resolves gh-263) 
fix: no metamodel tag fields generated for @Indexed Enums (resolves gh-277) 
fix: use ZoneId insteadof ZoneOffset to prevent 'Invalid ID for ZoneOffset, non numeric characters found: UTC' 
feature: app configuration properties for  EnhancedRedisRepository (resolves gh-268)
fix: addresses inconsistencies in query range generation for Date/Time types (resolves gh-270) 
feature: make DGL FeatureExtractor injectable (#262)
fix: handle GsonBuilder dynamic updates regardless of ClassLoader Brian Sam-Bodden
fix: correctly handle boolean returns in EntityStreams (resolved gh-265) (#269)
fix: handle deep nested returns (resolves gh-264 scenario 2) 
fix: allows null entries in Tuple returns (resolves gh-264) 
 feature: Implements @Reference annotation for JSON Document repositories and adapters 
refactor: rename reflection methods as to whether they return Type of Class
feature: add raw JSON.GET and JSON.MGET command methods
fix annotation processors in plugin as per the docs
```

### CODE REVIEW

File: redis-om-spring/src/main/java/com/redis/om/spring/repository/query/autocomplete/AutoCompleteOptions.java [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-
-@Setter
-@NoArgsConstructor
 public class AutoCompleteOptions {
   private boolean fuzzy = false;
   private int limit = 5;
   private boolean withScore = false;
   private boolean withPayload = false;
 
+  public AutoCompleteOptions() {
```
Comment (guyroyse) [inline redis-om-spring/src/main/java/com/redis/om/spring/repository/query/autocomplete/AutoCompleteOptions.java:9]: Is this constructor actually needed?

File: redis-om-spring/src/main/java/com/redis/om/spring/search/stream/AggregationStreamImpl.java [hunk 2]
```diff
@@ -40,12 +37,71 @@ public class AggregationStreamImpl<E, T> implements AggregationStream<T> {
...
+
+    public void setReducer( Reducer reducer) {
+      this.reducer = reducer;
+    }
+
+    public void setField(MetamodelField<?, ?> field) {
+      this.field = field;
+    }
+
+    public boolean equals(final Object o) {
```
Comment (guyroyse) [inline redis-om-spring/src/main/java/com/redis/om/spring/search/stream/AggregationStreamImpl.java:69]: This is why I hate Java!

**Review by guyroyse — APPROVED** [review]
One sarcastic comment, one optional thing that I thought looked odd, but good enough to merge as far as I'm concerned.

**Comment by bsbodden** [comment]
> One sarcastic comment, one optional thing that I thought looked odd, but good enough to merge as far as I'm concerned.

Yeah, I think the "DeLombification" automated tool process just added those, but no harm

**Review by guyroyse — DISMISSED** [review]
Ship it!

**Comment by bsbodden** [comment]
> Ship it!

Sorry, una vez mas porfavor. I forgot to rebase! 😬

**Review by guyroyse — APPROVED** [review]

**Pull request merged**